apps/Dataman/src/screens/Location.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState, useEffect } from "react";
+import { Text, View } from "react-native";
+import * as Location from "expo-location";
+
+export const LocationScreen = () => {
+  const [errorMsg, setErrorMsg] = useState("");
+  const [locationServiceEnabled, setLocationServiceEnabled] = useState(false);
```

Comment:
you should refactor this entire block of code into a custom hook

Reply:
refactored this code.